src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,241 @@
+    return {
+        "eventId": "Event Id",
+        "eventTimestatmp": "Timestamp",
+        "filePath": "Path",
+        "fileName": "File",
+        "exposure": "Exposure",
+    }
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def cases(state):
+    """For managing cases and events associated with cases."""
+    pass
+
+
+@cases.command()
+@click.argument("name")
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@sdk_options()
+def create(state, name, subject, assignee, description, notes):
+    """Create a new case."""
+    state.sdk.cases.create(
+        name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+    )
+
+
+@cases.command()
+@case_number_arg
+@name_option
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@status_option
+@sdk_options()
+def update(state, case_number, name, subject, assignee, description, notes, status):
+    """Update case details for the given case."""
+    state.sdk.cases.update(
+        case_number,
+        name=name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+        status=status,
+    )
+
+
+@cases.command("list")
+@click.option(
+    "--name", help="Filter by name of a case, supports partial name matches.",
+)
+@click.option("--assignee", required=False, help="Filter by user UID of assignee.")
+@click.option("--subject", help="Filter by user UID of the subject of a case.")
+@click.option("--assignee", help="Filter by user UID of assignee.")
+@click.option(
+    "--begin-create-time", help="Fetch cases created after given date time.",
+)
+@click.option(
+    "--end-create-time", help="Fetch cases created before given date time.",
+)
+@click.option(
+    "--begin-update-time", help="Fetch cases last updated after given date time.",
+)
+@click.option(
+    "--end-update-time", help="Fetch cases last updated before given date time.",
+)
+@click.option("--status", help="Filter cases by case status.")
+@format_option
+@sdk_options()
+def _list(
+    state,
+    name,
+    assignee,
+    subject,
+    begin_create_time,
+    end_create_time,
+    begin_update_time,
+    end_update_time,
+    status,
+    format,
+):
+    """List all the cases."""
+    pages = state.sdk.cases.get_all(
+        name=name,
+        assignee=assignee,
+        subject=subject,
+        min_create_time=begin_create_time,
+        max_create_time=end_create_time,
+        min_update_time=begin_update_time,
+        max_update_time=end_update_time,
+        status=status,
+    )
+    formatter = OutputFormatter(format, _get_cases_header())
+    cases = [case for page in pages for case in page["cases"]]
+    if cases:
+        formatter.echo_formatted_list(cases)
+    else:
+        click.echo("No cases found.")
+
+
+@cases.command()
+@case_number_arg
+@sdk_options()
+@format_option
+def show(state, case_number, format):
```

Comment:
I don't see the point in this command, it does not have any additional information.

I do not think we need a show command -- all of the data is already on `list`.

Reply:
Unlesssssss we have a `--include-events` flags and show the events associated with it too

Reply:
Similar to `code42 legal-hold show <id> --include-policy`

The events data can be JSON, as the policy in the legal hold command is also JSON.

So the small table and then the a header and then the json if want to include the event

Reply:
The json response does has more data, but I was not sure which fields are guaranteed and which to display so for `table` format, it is similar to `lists` command output, but for other formats it will contain more data.

Reply:
In list command, we can make `--file-events` option optional or rename it appropriately as suggested, but in either case we need to decide which fields are important and should be displayed in the table.

Reply:
Perhaps the file events JSON is too extreme (too large and noisy). Nevermind that! However, looking at the Ticket item for this: 

`
option exists for returning the full event data for events in the case
`

we should include all the data.  It is just 1 case. So, either:

1.) Create a table with all the data and then
1a.) Ignore capitalizing the header because we don't all the keys ahead of time and we have said in that past that is it not super important. There are even benefits to having the legitimate keys as header items as well.
1b.) If an expected header item is not on the response, just assume it is `None` and echo `None` for the row prop.

or

2.) Echo the full JSON blob. I'd go with this option if the JSON is very big (which I don't think it is) and ends up looking really bad as an ascii table